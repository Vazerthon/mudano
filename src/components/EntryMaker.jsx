import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import { DayPickerSingleDateController } from 'react-dates';

import { entryPropType } from './DayCell';

const Container = styled.div`
  display: flex;
`;

const StyledButton = styled.button`
  outline: none;
  border: none;
  width: ${({ double }) => (double ? '200px' : '100px')};
  margin: 2px;
  height: 20px;
  background-color: ${({ theme }) => theme.colour.primary};
  color: ${({ theme, on }) =>
    on ? theme.colour.highlight : theme.colour.secondary};
`;

const UnitContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const ValueContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const Selectors = styled.div`
  display: flex;
`;

const Buttons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const Warnings = styled.span``;

const allowSubmit = entry => entry.date && entry.unit && entry.value;

function EntryMaker({ stageNewEntry, entry, submitEntry, userId }) {
  const patch = diff => ({ ...entry, ...diff });

  return (
    <Container>
      <DayPickerSingleDateController
        numberOfMonths={3}
        onDateChange={date => stageNewEntry(patch({ date }))}
        date={entry.date}
      />

      <Buttons>
        <Selectors>
          <UnitContainer>
            <StyledButton
              on={entry.unit === 'AM'}
              onClick={() => stageNewEntry(patch({ unit: 'AM' }))}
            >
              AM
            </StyledButton>

            <StyledButton
              on={entry.unit === 'PM'}
              onClick={() => stageNewEntry(patch({ unit: 'PM' }))}
            >
              PM
            </StyledButton>
          </UnitContainer>

          <ValueContainer>
            <StyledButton
              on={entry.value === 'P'}
              onClick={() => stageNewEntry(patch({ value: 'P' }))}
            >
              Public Holiday
            </StyledButton>

            <StyledButton
              on={entry.value === 'V'}
              onClick={() => stageNewEntry(patch({ value: 'V' }))}
            >
              Vacation
            </StyledButton>

            <StyledButton
              on={entry.value === 'T'}
              onClick={() => stageNewEntry(patch({ value: 'T' }))}
            >
              Training
            </StyledButton>

            <StyledButton
              on={entry.value === 'PRESENT'}
              onClick={() => stageNewEntry(patch({ value: 'PRESENT' }))}
            >
              Present
            </StyledButton>
          </ValueContainer>
        </Selectors>
        <Warnings>TODO - warning text</Warnings>
        <StyledButton
          on={allowSubmit(entry)}
          disabled={!allowSubmit(entry)}
          double
          onClick={() => submitEntry(userId, entry)}
        >
          Sumbit
        </StyledButton>
      </Buttons>
    </Container>
  );
}

EntryMaker.propTypes = {
  stageNewEntry: PropTypes.func.isRequired,
  entry: entryPropType,
  submitEntry: PropTypes.func.isRequired,
  userId: PropTypes.string,
};

EntryMaker.defaultProps = {
  entry: {},
  userId: '',
};

export default EntryMaker;
