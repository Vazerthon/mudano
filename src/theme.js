const theme = {
  colour: {
    primary: '#1e252b',
    secondary: '#dcd6ca',
    highlight: '#9a3cbb',
    timeline: {
      P: '#0971b2', // public holiday
      V: '#54cc14', // vacation
      T: '#cc4a14', // training
      present: '#9a3cbb',
    },
  },
  layout: {
    padding: '8px',
  },
};

export default theme;
