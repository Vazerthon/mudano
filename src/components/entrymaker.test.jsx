/* eslint-env jest */

import React from 'react';
import shallowThemeWrapper from './test-helper';
import theme from '../theme';

import EntryMaker from './EntryMaker';

const shallow = component => shallowThemeWrapper(component, theme);

describe('EntryMaker component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EntryMaker stageNewEntry={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
