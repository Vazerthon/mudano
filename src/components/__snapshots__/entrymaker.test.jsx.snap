// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryMaker component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EntryMaker
    entry={Object {}}
    stageNewEntry={[Function]}
    submitEntry={[Function]}
    userId=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DayPickerSingleDateController
          date={undefined}
          daySize={39}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isFocused={false}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          noBorder={false}
          numberOfMonths={3}
          onBlur={[Function]}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onOutsideClick={null}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
                    Object {
                              "calendarLabel": "Calendar",
                              "chooseAvailableDate": [Function],
                              "chooseAvailableEndDate": [Function],
                              "chooseAvailableStartDate": [Function],
                              "dateIsSelected": [Function],
                              "dateIsUnavailable": [Function],
                              "enterKey": "Enter key",
                              "escape": "Escape key",
                              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                              "homeEnd": "Home and end keys",
                              "jumpToNextMonth": "Move forward to switch to the next month.",
                              "jumpToPrevMonth": "Move backward to switch to the previous month.",
                              "keyboardShortcuts": "Keyboard Shortcuts",
                              "leftArrowRightArrow": "Right and left arrow keys",
                              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                              "moveFocusByOneMonth": "Switch months.",
                              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                              "openThisPanel": "Open this panel.",
                              "pageUpPageDown": "page up and page down keys",
                              "questionMark": "Question mark",
                              "returnFocusToInput": "Return to the date input field.",
                              "selectFocusedDate": "Select the date in focus.",
                              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                              "upArrowDownArrow": "up and down arrow keys",
                            }
          }
          renderCalendarDay={undefined}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonth={null}
          showKeyboardShortcuts={false}
          transitionDuration={undefined}
          verticalHeight={null}
          weekDayFormat="dd"
          withPortal={false}
/>,
        <styled.div>
          <styled.div>
                    <styled.div>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        AM
                              </styled.button>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        PM
                              </styled.button>
                    </styled.div>
                    <styled.div>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        Public Holiday
                              </styled.button>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        Vacation
                              </styled.button>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        Training
                              </styled.button>
                              <styled.button
                                        on={false}
                                        onClick={[Function]}
                              >
                                        Present
                              </styled.button>
                    </styled.div>
          </styled.div>
          <styled.span>
                    TODO - warning text
          </styled.span>
          <styled.button
                    disabled={true}
                    double={true}
                    on={undefined}
                    onClick={[Function]}
          >
                    Sumbit
          </styled.button>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "date": undefined,
          "daySize": 39,
          "enableOutsideDays": false,
          "firstDayOfWeek": null,
          "focused": false,
          "hideKeyboardShortcutsPanel": false,
          "initialVisibleMonth": null,
          "isDayBlocked": [Function],
          "isDayHighlighted": [Function],
          "isFocused": false,
          "isOutsideRange": [Function],
          "isRTL": false,
          "keepOpenOnDateSelect": false,
          "monthFormat": "MMMM YYYY",
          "navNext": null,
          "navPrev": null,
          "noBorder": false,
          "numberOfMonths": 3,
          "onBlur": [Function],
          "onClose": [Function],
          "onDateChange": [Function],
          "onFocusChange": [Function],
          "onNextMonthClick": [Function],
          "onOutsideClick": null,
          "onPrevMonthClick": [Function],
          "orientation": "horizontal",
          "phrases": Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "chooseAvailableEndDate": [Function],
            "chooseAvailableStartDate": [Function],
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          },
          "renderCalendarDay": undefined,
          "renderCalendarInfo": null,
          "renderDayContents": null,
          "renderMonth": null,
          "showKeyboardShortcuts": false,
          "transitionDuration": undefined,
          "verticalHeight": null,
          "weekDayFormat": "dd",
          "withPortal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.div>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          AM
                            </styled.button>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          PM
                            </styled.button>
              </styled.div>
              <styled.div>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          Public Holiday
                            </styled.button>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          Vacation
                            </styled.button>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          Training
                            </styled.button>
                            <styled.button
                                          on={false}
                                          onClick={[Function]}
                            >
                                          Present
                            </styled.button>
              </styled.div>
</styled.div>,
            <styled.span>
              TODO - warning text
</styled.span>,
            <styled.button
              disabled={true}
              double={true}
              on={undefined}
              onClick={[Function]}
>
              Sumbit
</styled.button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    AM
                  </styled.button>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    PM
                  </styled.button>
</styled.div>,
                <styled.div>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    Public Holiday
                  </styled.button>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    Vacation
                  </styled.button>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    Training
                  </styled.button>
                  <styled.button
                                    on={false}
                                    onClick={[Function]}
                  >
                                    Present
                  </styled.button>
</styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      AM
</styled.button>,
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      PM
</styled.button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "AM",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "AM",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "PM",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "PM",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      Public Holiday
</styled.button>,
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      Vacation
</styled.button>,
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      Training
</styled.button>,
                    <styled.button
                      on={false}
                      onClick={[Function]}
>
                      Present
</styled.button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Public Holiday",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Public Holiday",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Vacation",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Vacation",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Training",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Training",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Present",
                      "on": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Present",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "TODO - warning text",
            },
            "ref": null,
            "rendered": "TODO - warning text",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sumbit",
              "disabled": true,
              "double": true,
              "on": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sumbit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DayPickerSingleDateController
            date={undefined}
            daySize={39}
            enableOutsideDays={false}
            firstDayOfWeek={null}
            focused={false}
            hideKeyboardShortcutsPanel={false}
            initialVisibleMonth={null}
            isDayBlocked={[Function]}
            isDayHighlighted={[Function]}
            isFocused={false}
            isOutsideRange={[Function]}
            isRTL={false}
            keepOpenOnDateSelect={false}
            monthFormat="MMMM YYYY"
            navNext={null}
            navPrev={null}
            noBorder={false}
            numberOfMonths={3}
            onBlur={[Function]}
            onClose={[Function]}
            onDateChange={[Function]}
            onFocusChange={[Function]}
            onNextMonthClick={[Function]}
            onOutsideClick={null}
            onPrevMonthClick={[Function]}
            orientation="horizontal"
            phrases={
                        Object {
                                    "calendarLabel": "Calendar",
                                    "chooseAvailableDate": [Function],
                                    "chooseAvailableEndDate": [Function],
                                    "chooseAvailableStartDate": [Function],
                                    "dateIsSelected": [Function],
                                    "dateIsUnavailable": [Function],
                                    "enterKey": "Enter key",
                                    "escape": "Escape key",
                                    "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                                    "homeEnd": "Home and end keys",
                                    "jumpToNextMonth": "Move forward to switch to the next month.",
                                    "jumpToPrevMonth": "Move backward to switch to the previous month.",
                                    "keyboardShortcuts": "Keyboard Shortcuts",
                                    "leftArrowRightArrow": "Right and left arrow keys",
                                    "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                                    "moveFocusByOneMonth": "Switch months.",
                                    "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                                    "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                                    "openThisPanel": "Open this panel.",
                                    "pageUpPageDown": "page up and page down keys",
                                    "questionMark": "Question mark",
                                    "returnFocusToInput": "Return to the date input field.",
                                    "selectFocusedDate": "Select the date in focus.",
                                    "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                                    "upArrowDownArrow": "up and down arrow keys",
                                  }
            }
            renderCalendarDay={undefined}
            renderCalendarInfo={null}
            renderDayContents={null}
            renderMonth={null}
            showKeyboardShortcuts={false}
            transitionDuration={undefined}
            verticalHeight={null}
            weekDayFormat="dd"
            withPortal={false}
/>,
          <styled.div>
            <styled.div>
                        <styled.div>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                AM
                                    </styled.button>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                PM
                                    </styled.button>
                        </styled.div>
                        <styled.div>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                Public Holiday
                                    </styled.button>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                Vacation
                                    </styled.button>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                Training
                                    </styled.button>
                                    <styled.button
                                                on={false}
                                                onClick={[Function]}
                                    >
                                                Present
                                    </styled.button>
                        </styled.div>
            </styled.div>
            <styled.span>
                        TODO - warning text
            </styled.span>
            <styled.button
                        disabled={true}
                        double={true}
                        on={undefined}
                        onClick={[Function]}
            >
                        Sumbit
            </styled.button>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "date": undefined,
            "daySize": 39,
            "enableOutsideDays": false,
            "firstDayOfWeek": null,
            "focused": false,
            "hideKeyboardShortcutsPanel": false,
            "initialVisibleMonth": null,
            "isDayBlocked": [Function],
            "isDayHighlighted": [Function],
            "isFocused": false,
            "isOutsideRange": [Function],
            "isRTL": false,
            "keepOpenOnDateSelect": false,
            "monthFormat": "MMMM YYYY",
            "navNext": null,
            "navPrev": null,
            "noBorder": false,
            "numberOfMonths": 3,
            "onBlur": [Function],
            "onClose": [Function],
            "onDateChange": [Function],
            "onFocusChange": [Function],
            "onNextMonthClick": [Function],
            "onOutsideClick": null,
            "onPrevMonthClick": [Function],
            "orientation": "horizontal",
            "phrases": Object {
              "calendarLabel": "Calendar",
              "chooseAvailableDate": [Function],
              "chooseAvailableEndDate": [Function],
              "chooseAvailableStartDate": [Function],
              "dateIsSelected": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            },
            "renderCalendarDay": undefined,
            "renderCalendarInfo": null,
            "renderDayContents": null,
            "renderMonth": null,
            "showKeyboardShortcuts": false,
            "transitionDuration": undefined,
            "verticalHeight": null,
            "weekDayFormat": "dd",
            "withPortal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.div>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                AM
                                </styled.button>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                PM
                                </styled.button>
                </styled.div>
                <styled.div>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                Public Holiday
                                </styled.button>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                Vacation
                                </styled.button>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                Training
                                </styled.button>
                                <styled.button
                                                on={false}
                                                onClick={[Function]}
                                >
                                                Present
                                </styled.button>
                </styled.div>
</styled.div>,
              <styled.span>
                TODO - warning text
</styled.span>,
              <styled.button
                disabled={true}
                double={true}
                on={undefined}
                onClick={[Function]}
>
                Sumbit
</styled.button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        AM
                    </styled.button>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        PM
                    </styled.button>
</styled.div>,
                  <styled.div>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        Public Holiday
                    </styled.button>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        Vacation
                    </styled.button>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        Training
                    </styled.button>
                    <styled.button
                                        on={false}
                                        onClick={[Function]}
                    >
                                        Present
                    </styled.button>
</styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        AM
</styled.button>,
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        PM
</styled.button>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "AM",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "AM",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "PM",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "PM",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        Public Holiday
</styled.button>,
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        Vacation
</styled.button>,
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        Training
</styled.button>,
                      <styled.button
                        on={false}
                        onClick={[Function]}
>
                        Present
</styled.button>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Public Holiday",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Public Holiday",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Vacation",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Vacation",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Training",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Training",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Present",
                        "on": false,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Present",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "TODO - warning text",
              },
              "ref": null,
              "rendered": "TODO - warning text",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sumbit",
                "disabled": true,
                "double": true,
                "on": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sumbit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "__styled-components__": [Function],
      "__styled-components__next__": Object {
        "getTheme": [Function],
        "subscribe": [Function],
        "unsubscribe": [Function],
      },
    },
  },
}
`;
